es_ES/faq.md

```diff
@@ -0,0 +1,625 @@
+
+* Programación genérica: los tipos pueden usarse para ayudar a los desarrolladores a captar errores tempranos en la programación. El framework de Flutter utiliza programación genérica para ayudar en esto. Por ejemplo, la clase [State](https://docs.flutter.io/flutter/widgets/State-class.html) es parametrizada en términos del tipo de su widget asociado, de modo que el analizador Dart puede detectar desajustes entre estados y widgets. Del mismo modo, la clase [GlobalKey](https://docs.flutter.io/flutter/widgets/GlobalKey-class.html) toma un parámetro de tipo para que pueda acceder al estado de un widget remoto en un tipo de manera segura (utilizando la comprobación de tiempo de ejecución), la interfaz [Route](https://docs.flutter.io/flutter/widgets/Route-class.html) es parametrizada con el tipo que se espera que use cuando [popped](https://docs.flutter.io/flutter/widgets/Navigator/pop.html), y colecciones tales como [List](https://docs.flutter.io/flutter/dart-core/List-class.html), [Map](https://docs.flutter.io/flutter/dart-core/Map-class.html), y [Set](https://docs.flutter.io/flutter/dart-core/Set-class.html) todas son parametrizadas de modo que los elementos no coincidentes se puedan detectar temprano durante análisis o en tiempo de ejecución durante la depuración.
+
+* Programación concurrente: Flutter hace un uso intensivo de [Future](https://docs.flutter.io/flutter/dart-async/Future-class.html) y otras API asíncronas. Por ejemplo, el sistema de animación informa cuando la animación se completa al completar un futuro. El sistema de carga de imagen de manera similar utiliza futuros para informar cuando se completa una carga.
+
+* Programación de restricción: el sistema de disposición en Flutter usa una forma débil de programación de restricciones para determinar la geometría de una escena. Restricciones (para cajas cartesianas, un ancho mínimo y máximo y un mínimo y máximo altura) se pasan de padres a hijos, y el hijo selecciona una geometría resultante (por ejemplo, para cajas cartesianas, un tamaño, específicamente un ancho y una altura) que cumple con esas restricciones. Al usar esta técnica, Flutter generalmente puede diseñar una escena completa con un solo pase.
+
+## Proyecto
+
+### ¿Donde puedo obtener soporte?
+
+Si crees que has encontrado un error, por favor archívalo en nuestro
+[rastreador de problemas](https://github.com/flutter/flutter/issues). Nosotros te animamos a usar [Stack Overflow](https://stackoverflow.com/tags/flutter) para preguntas tipo "HOWTO". Para discusiones, por favor, únete a nuestra lista de correo en [flutter-dev@googlegroups.com](mailto:flutter-dev@googlegroups.com).
+
+### ¿Cómo me involucro?
+
+Flutter es de código abierto, y te animamos a contribuir. Puedes comenzar por
```

Comment:
You can start by simply filing issues for feature requests and bugs in our issue tracker.
Puedes empezar simplemente rellenando issues de bugs o requiriendo funcionalidades en nuestro issue tracker.